core/trino-main/src/main/java/io/trino/sql/planner/planprinter/DistributedFragmentRepresentation.java

```diff
@@ -0,0 +1,495 @@
+        List<NodeRepresentation> children = node.getChildren().stream()
+                .map(plan::getNode)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(toList());
+        List<DistributedFragmentRepresentation> childrenRepresentation = new ArrayList<>();
+        for (NodeRepresentation child : children) {
+            childrenRepresentation.add(getPlanRepresentation(plan, child, verbose, level + 1));
+        }
+
+        Optional<PlanNodeStats> nodeStats = node.getStats();
+        Optional<WindowOperatorStatsRepresentation> windowOperatorStatsRepresentation = Optional.empty();
+        if (nodeStats.isPresent() && (nodeStats.get() instanceof WindowPlanNodeStats) && verbose) {
+            WindowOperatorStats windowOperatorStats = ((WindowPlanNodeStats) nodeStats.get()).getWindowOperatorStats();
+            windowOperatorStatsRepresentation = Optional.of(new WindowOperatorStatsRepresentation(windowOperatorStats));
+        }
+
+        return new DistributedFragmentRepresentation(
+                node.getName(),
+                node.getIdentifier(),
+                node.getOutputs(),
+                node.getDetails(),
+                planNodeStatsAndCostSummary,
+                estimateStats,
+                distributedFragmentStats,
+                childrenRepresentation,
+                windowOperatorStatsRepresentation,
+                node);
+    }
+
+    private static List<EstimateStats> getEstimates(PlanRepresentation plan, NodeRepresentation node)
+    {
+        if (node.getEstimatedStats().stream().allMatch(PlanNodeStatsEstimate::isOutputRowCountUnknown) &&
+                node.getEstimatedCost().stream().allMatch(c -> c.equals(PlanCostEstimate.unknown()))) {
+            return new ArrayList<>();
+        }
+
+        List<EstimateStats> estimateStats = new ArrayList<>();
+        int estimateCount = node.getEstimatedStats().size();
+
+        for (int i = 0; i < estimateCount; i++) {
+            PlanNodeStatsEstimate stats = node.getEstimatedStats().get(i);
+            PlanCostEstimate cost = node.getEstimatedCost().get(i);
+
+            List<Symbol> outputSymbols = node.getOutputs().stream()
+                    .map(NodeRepresentation.TypedSymbol::getSymbol)
+                    .collect(toList());
+
+            estimateStats.add(new EstimateStats(
+                    formatAsLong(stats.getOutputRowCount()),
+                    formatAsDataSize(stats.getOutputSizeInBytes(outputSymbols, plan.getTypes())),
+                    formatAsCpuCost(cost.getCpuCost()),
+                    formatAsDataSize(cost.getMaxMemory()),
+                    formatAsDataSize(cost.getNetworkCost())));
+        }
+
+        return estimateStats;
+    }
+
+    private static Optional<PlanNodeStatsAndCostSummary> getReorderJoinStatsAndCost(NodeRepresentation node, boolean verbose)
+    {
+        if (verbose && node.getReorderJoinStatsAndCost().isPresent()) {
```

Comment:
`verbose` should be a display-time concern. The representation object should contain all the relevant data, and the logic that prints the plan should choose what to include or exclude based on the verbose flag.

Reply:
Since I use JsonCodec, depending on whether `verbose` is present or not, I am creating the `windowOperatorStats` object. The JsonRenderer will just take the object and render using JsonCodec. Determining whether to render a member of class depending on whether `verbose` is present or not will make rendering more complex. I am not sure how to even do that actually with JsonCodec.